Training6/task2/src/redux/saga/network.js

```diff
@@ -0,0 +1,38 @@
+import { take, put, fork, delay } from "redux-saga/effects";
+import { eventChannel } from "redux-saga";
+import { syncNetworkStatus } from "../actions/network";
+
+function* startChannelNetwork() {
+  // Move the network status listener to redux-saga using even channel
+  const channelNetwork = eventChannel((listener) => {
+    window.addEventListener("offline", function () {
+      listener(false);
+    });
+
+    window.addEventListener("online", function () {
+      listener(true);
+    });
+
+    return () => {
+      window.removeEventListener("online");
+      window.removeEventListener("offline");
+    };
+  });
+
+  while (true) {
+    const connectionInfo = yield take(channelNetwork);
+    yield put(syncNetworkStatus(connectionInfo));
+  }
+}
+
+function* networkSaga() {
+  try {
+    yield delay(500);
```

Comment:
@thienvu23 why we need delay here

Reply:
I don't think we need to delay and put action network status here, you can set default navigator.onLine on reducer I commented above